SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp

```diff
@@ -28,170 +30,148 @@ void USpatialTypebindingManager::Init(USpatialNetDriver* InNetDriver)
-				for (int32 ArrayIdx = 0; ArrayIdx < PropertyIt->ArrayDim; ++ArrayIdx)
-				{
-					FInterestPropertyInfo InterestInfo;
-					InterestInfo.Offset = Property->GetOffset_ForGC() + Property->ElementSize * ArrayIdx;
-					InterestInfo.Property = Property;
-
-					Info.InterestProperties.Add(InterestInfo);
-				}
-			}
-		}
+		SubobjectInfo.SubobjectName = SubobjectSchemaData.Name;
 
 		ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
 		{
-			Worker_ComponentId ComponentId = SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SchemaComponents[Type];
+			Worker_ComponentId ComponentId = SubobjectSchemaData.SchemaComponents[Type];
 			if (ComponentId != 0)
 			{
-				Info.SchemaComponents[Type] = ComponentId;
-				ComponentToClassMap.Add(ComponentId, Class);
-				ComponentToOffsetMap.Add(ComponentId, 0);
+				SubobjectInfo.SchemaComponents[Type] = ComponentId;
+				ComponentToClassMap.Add(ComponentId, SubobjectClass);
+				ComponentToOffsetMap.Add(ComponentId, Offset);
 				ComponentToCategoryMap.Add(ComponentId, (ESchemaComponentType)Type);
 			}
 		});
 
-		Info.Class = Class;
-
-		ClassInfoMap.Emplace(Class, Info);
+		Info.SubobjectInfo.Add(Offset, MakeShared<FClassInfo>(SubobjectInfo));
 	}
 
-	for (UClass* Class : SupportedClasses)
-	{
-		for (auto& SubobjectDataPair : SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SubobjectData)
-		{
-			int32 Offset = SubobjectDataPair.Key;
-			FSubobjectSchemaData SubobjectSchemaData = SubobjectDataPair.Value;
-
-			FClassInfo* ActorInfo = FindClassInfoByClass(Class);
-
-			FSoftClassPath SubobjectClassPath(SubobjectSchemaData.ClassPath);
-			UClass* SubobjectClass = SubobjectClassPath.TryLoadClass<UObject>();
-			if (SubobjectClass == nullptr)
-			{
-				continue;
-			}
-
-			FClassInfo* SubobjectInfoPtr = FindClassInfoByClass(SubobjectClass);
-			if (SubobjectInfoPtr == nullptr)
-			{
-				continue;
-			}
-
-			// Make a copy of the already made FClassInfo for this specific subobject
-			FClassInfo SubobjectInfo = *SubobjectInfoPtr;
-
-			SubobjectInfo.SubobjectName = SubobjectSchemaData.Name;
-
-			ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
-			{
-				Worker_ComponentId ComponentId = SubobjectSchemaData.SchemaComponents[Type];
-				if (ComponentId != 0)
-				{
-					SubobjectInfo.SchemaComponents[Type] = ComponentId;
-					ComponentToClassMap.Add(ComponentId, SubobjectClass);
-					ComponentToOffsetMap.Add(ComponentId, Offset);
-					ComponentToCategoryMap.Add(ComponentId, (ESchemaComponentType)Type);
-				}
-			});
-
-			ActorInfo->SubobjectInfo.Add(Offset, MakeShared<FClassInfo>(SubobjectInfo));
-		}
-	}
+	return Info;
 }
 
 FClassInfo* USpatialTypebindingManager::FindClassInfoByClass(UClass* Class)
 {
-	return ClassInfoMap.Find(Class);
+	if (!IsSupportedClass(Class))
+	{
+		UE_LOG(LogTypebindingManager, Warning, TEXT("Could not find class in schema database: %s"), *Class->GetPathName());
+		return nullptr;
+	}
+
+	if (FClassInfo* Info = ClassInfoMap.Find(Class))
```

Comment:
```suggestion
	if (!ClassInfoMap.Contains(Class))
	{
		AddTypebindingsForClass(Class);
	}
	
	return ClassInfoMap.Find(Class);
```

Reply:
It was like this in my first implementation, but this does two searches in the map - one for Contains and one for Find.

Reply:
Discussed offline and will accept the suggestion in favour of readability.